### PR TITLE
Week1/Assignment - How Web Works

### PR DESCRIPTION
**Pull request by mrkarthik3**

### CODE REVIEW

File: Week-1/how-web-works/how-web-works.md [hunk 1]
```diff
@@ -0,0 +1,116 @@
+## What is the main functionality of a browser ? ##
+
+A browser uses a URI (Uniform Resource Identifier) and sends requests for data(HTML/CSS/JS/XML/JSON and others) to other nodes(server/another computer) in its network and serve received data to the requested user.
```
Comment (sairamin) [inline Week-1/how-web-works/how-web-works.md]: What is its network ?

File: Week-1/how-web-works/how-web-works.md [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+7. Data Storage:
+
+This is where data is stored locally for persistence (Eg. cookies)
+Browsers support storage mechanisms such as localStorage, IndexedDB, WebSQL and FileSystem
+
+## Rendering Engine and its Use ##
+
+![webkit rendering engine](./images/webkitflow-renderingengine.png)
+
+Rendering engine interacts with network layer to get documents(HTML/CSS etc) in chunks of 8kbs.
```
Comment (sairamin) [inline Week-1/how-web-works/how-web-works.md:50]: Can you provide documentation that supports this ?

File: Week-1/how-web-works/how-web-works.md [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+Thus rendering engine parses HTML and CSS files.
+Each html element is convered to DOM node. 
+A Content tree/DOM tree containing DOM nodes is thus constructured.
```
Comment (sairamin) [inline Week-1/how-web-works/how-web-works.md]: constructured ?

File: Week-1/how-web-works/how-web-works.md [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+
+Simultaneously all style data is parsed, and a "Render tree" is created.
```
Comment (sairamin) [inline Week-1/how-web-works/how-web-works.md]: Wrong

**Review by sairamin — CHANGES_REQUESTED** [review]
Read through the articles that are shared and update the findings.

File: Week-1/how-web-works/how-web-works.md (hunk 1 above)
Comment (mrkarthik3) [reply Week-1/how-web-works/how-web-works.md]: A browser can request files served by a server in local network (intranet) or the internet. I made the distinction between them. Hence I chose the word "network".

File: Week-1/how-web-works/how-web-works.md (hunk 2 above)
Comment (mrkarthik3) [reply Week-1/how-web-works/how-web-works.md:50]: Yes.  https://www.html5rocks.com/en/tutorials/internals/howbrowserswork/#The_main_flow
Added this link at the bottom.

File: Week-1/how-web-works/how-web-works.md (hunk 3 above)
Comment (mrkarthik3) [reply Week-1/how-web-works/how-web-works.md]: It was a typo! Corrected it.

File: Week-1/how-web-works/how-web-works.md (hunk 4 above)
Comment (mrkarthik3) [reply Week-1/how-web-works/how-web-works.md]: CSS is parsed and CSSOM Tree is created. Corrected the mistake.

**Review by sairamin — APPROVED** [review]

**Pull request merged**